geometry.go

```diff
@@ -49,6 +49,15 @@ func (u Vec) XY() (x, y float64) {
 	return u.X, u.Y
 }
 
+// unit vector of u, rotated by math.pi/2
+func (u Vec) Normal() Vec {
+	l := math.Hypot(u.X, u.Y)
+	if l == 0 {
+		l = 1
+	}
+	return Vec{X: u.Y / l, Y: -u.X / l}
+}
+
```

Comment:
Ok, all is good except one thing. This implementation is inconsistent with Unit(). On a zero sized vector, Unit() returns (1,0), but this returns (0,0). No good, this should return (0,1) in such case.

Also, is using u.Len() here too expensive?